erpnext/stock/doctype/batch/batch.js

```diff
@@ -22,8 +22,18 @@ frappe.ui.form.on('Batch', {
 				frappe.set_route("query-report", "Stock Ledger");
 			});
 			frm.trigger('make_dashboard');
+			frm.add_custom_button(__('Make Material Request'), () => {
+				frm.trigger("make_request")
+			});
 		}
 	},
+	make_request: function () {
+		frappe.model.open_mapped_doc({
+			method: "erpnext.stock.doctype.material_request.material_request.make_material_request",
+			frm: cur_frm
```

Comment:
Does using `frm` work here?

```suggestion
			frm: frm
```